src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -1446,7 +3417,90 @@ .maxstack  1
     }
 
     [Fact]
-    public void Conversion_Array_Span_Implicit_Params()
+    public void Conversion_Span_ReadOnlySpan_Opposite_Explicit_UserDefined()
+    {
+        var source = """
+            using System;
+            class C
+            {
+                 Span<int> M(ReadOnlySpan<int> arg) => (Span<int>)arg;
+            }
+
+            namespace System
+            {
+                readonly ref struct Span<T>;
+                readonly ref struct ReadOnlySpan<T>
+                {
+                    public static explicit operator Span<T>(ReadOnlySpan<T> span) => throw null;
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(source, parseOptions: TestOptions.Regular13, verify: Verification.FailsILVerify);
+        verifier.VerifyDiagnostics();
+        verifier.VerifyIL("C.M", """
+            {
+              // Code size        7 (0x7)
+              .maxstack  1
+              IL_0000:  ldarg.1
+              IL_0001:  call       "System.Span<int> System.ReadOnlySpan<int>.op_Explicit(System.ReadOnlySpan<int>)"
+              IL_0006:  ret
+            }
+            """);
+
+        var expectedDiagnostics = new[]
+        {
+            // (4,44): error CS0030: Cannot convert type 'System.ReadOnlySpan<int>' to 'System.Span<int>'
+            //      Span<int> M(ReadOnlySpan<int> arg) => (Span<int>)arg;
+            Diagnostic(ErrorCode.ERR_NoExplicitConv, "(Span<int>)arg").WithArguments("System.ReadOnlySpan<int>", "System.Span<int>").WithLocation(4, 44)
```

Comment:
I would expect this to be legal.

Reply:
Why? There is no implicit nor explicit conversion from ReadOnlySpan to Span. There is an explicit span conversion only from array to (ReadOnly)Span. Similarly we could propose other explicit conversions but I'm not sure how we would emit them anyway.

Reply:
Because the type defines a UDC that permits it.

Reply:
Ah I missed that. Then you're right that looks a bit unexpected, let me check what's happening.

Reply:
Okay, this UDC is ignored - per [the speclet](https://github.com/dotnet/csharplang/blob/main/proposals/first-class-span-types.md#user-defined-conversions):

> User-defined conversions are not considered when converting between
> - any single-dimensional `array_type` and `System.Span<T>`/`System.ReadOnlySpan<T>`,
> - any combination of `System.Span<T>`/`System.ReadOnlySpan<T>`,
> - `string` and `System.ReadOnlySpan<char>`.

This case is the second bullet point.

Reply:
I'm not sure why the speclet says that. The only things the speclet should talk about are the conversions that it actually cares about. It does not care about converting from `ReadOnlySpan` to `Span.

Reply:
The spec should certainly say that no ROS->Span standard explicit conversion exists, but that isn't what we have here. We have a UDC. We shouldn't be disallowing this, as the feature doesn't care about it. It of course can't fall in the bucket of a standard implicit conversion; it can't contribute to the things that only "standard" conversions can. But it can absolutely still exist and be used, and be called by user code as any UDC can.

Reply:
Okay, works for me. Updating the speclet in https://github.com/dotnet/csharplang/pull/8312 (the second commit is relevant to this discussion).

Reply:
> I'm not sure why the speclet says that. The only things the speclet should talk about are the conversions that it actually cares about. It does not care about converting from `ReadOnlySpan` to `Span`.

My expectation is that if some conversions between spans and arrays, spans, and strings are handled directly in the compiler, then all such conversions should be handled in the compiler, and the compiler should not fallback to user-defined conversions between these types. Was this covered at LDM?

Reply:
> Was this covered at LDM?

It wasn't, adding an open question in https://github.com/dotnet/csharplang/pull/8312.